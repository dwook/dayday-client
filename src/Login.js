import React, {Component} from 'react';
import {
  View,
  Text,
  StyleSheet,
  StatusBar,
  TextInput,
  AsyncStorage,
  ActivityIndicator,
} from 'react-native';

import Touchable from '@appandflow/touchable';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import {LoginManager, AccessToken} from 'react-native-fbsdk';

import {authToken} from '../../utils/constants';
import {startMainApp} from '../../Nav';

const styles = StyleSheet.create({});

class LoginScreen extends Component {
  state = {
    loading: false,
  };

  _onLoginFbPress = async () => {
    this.setState({loading: true});

    let res;
    try {
      res = await LoginManager.logInWithReadPermissions([
        'public_profile',
        'email',
      ]);
    } catch (error) {
      console.log('====================================');
      console.log('error', error);
      console.log('====================================');
    }

    if (res.grantedPermissions && !res.isCancelled) {
      const data = await AccessToken.getCurrentAccessToken();

      if (data) {
        const serverResponse = await this.props.loginMutation({
          variables: {
            provider: 'FACEBOOK',
            token: data.accessToken,
          },
        });

        const {token} = serverResponse.data.login;

        try {
          await AsyncStorage.setItem(authToken, token);

          this.setState({loading: false});

          startMainApp();
        } catch (error) {
          throw error;
        }
      }
    }
  };

  render() {
    if (this.state.loading) {
      return (
        <View style={styles.root}>
          <ActivityIndicator size="large" color="#318DEE" />
        </View>
      );
    }
    return (
      <View style={styles.root}>
        <Text>Continue with Facebook</Text>
      </View>
    );
  }
}

const loginMutation = gql`
  mutation($provider: Provider, $token: String) {
    login(provider: $provider, token: $token) {
      token
    }
  }
`;

export default graphql(loginMutation, {name: 'loginMutation'})(LoginScreen);
