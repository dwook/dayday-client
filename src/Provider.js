import React, {Component} from 'react';
import {AsyncStorage} from 'react-native';
import axios from 'axios';

export const AppContext = React.createContext();

export default class AppProvider extends Component {
  state = {
    token: '',
    user: {
      id: '',
      name: '',
      profile_image: '',
    },
    diary: {
      good: 'a',
      bad: 'a',
      plan: 'a',
    },
    setUser: user => {
      this.setState({
        user: {
          id: user._id,
          name: user.name,
          profile_image: user.profile_image,
        },
      });
      console.log('등록된유저정보', this.state.user);
    },
    saveToken: async userToken => {
      try {
        await AsyncStorage.setItem('userToken', userToken);
        console.log('유저토큰222', userToken);
      } catch (error) {
        this.setState({error});
      }
    },
    enterText: async text => {
      await this.setState(currentState => {
        return {
          ...currentState,
          diary: {
            ...currentState.diary,
            good: text,
          },
        };
      });
      console.log('내용작성', this.state.diary);
    },
    sendDiary: async () => {
      await axios
        .post('http://localhost:3000/diaries', {diary: this.state.diary})
        .catch(err => {
          console.log(err);
        });
    },
    removeToken: async () => {
      try {
        await AsyncStorage.removeItem('userToken');
      } catch (error) {
        this.setState({error});
      }
    },
    getToken: async () => {
      try {
        await AsyncStorage.getItem('userToken');
      } catch (error) {
        this.setState({error});
      }
    },
  };

  render() {
    return (
      <AppContext.Provider value={this.state}>
        {this.props.children}
      </AppContext.Provider>
    );
  }
}
