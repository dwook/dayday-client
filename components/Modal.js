import React from 'react';
import {Animated, TouchableOpacity, Dimensions, View} from 'react-native';
import {AppContext} from '../src/Provider';
import styled from 'styled-components/native';
import {CloseIcon} from './Icons';
import {THEME_COLOR} from '../constant';

const screenWidth = Dimensions.get('window').width;
const screenHeight = Dimensions.get('window').height;

export default class Modal extends React.Component {
  render() {
    return (
      <AppContext.Consumer>
        {context => (
          <AnimatedContainer style={{top: context.modalTop}}>
            <CloseIconWrap>
              <TouchableOpacity
                onPress={() => {
                  context.toggleModal('close');
                }}>
                <CloseIcon stroke={'#000'} />
              </TouchableOpacity>
            </CloseIconWrap>
            <ContentWrap>
              <Content>
                <TitleWrap>
                  <Dot color={THEME_COLOR.good} />
                  <TitleText color={THEME_COLOR.good}> GOOD</TitleText>
                </TitleWrap>
                <DiaryText>{context.selectedDiary.good}</DiaryText>
              </Content>
              <Content>
                <TitleWrap>
                  <Dot color={THEME_COLOR.bad} />
                  <TitleText color={THEME_COLOR.bad}> BAD</TitleText>
                </TitleWrap>
                <DiaryText>{context.selectedDiary.bad}</DiaryText>
              </Content>
              <Content>
                <TitleWrap>
                  <Dot color={THEME_COLOR.plan} />
                  <TitleText color={THEME_COLOR.plan}> PLAN</TitleText>
                </TitleWrap>
                <DiaryText>{context.selectedDiary.plan}</DiaryText>
              </Content>
            </ContentWrap>
          </AnimatedContainer>
        )}
      </AppContext.Consumer>
    );
  }
}

const Container = styled.View`
  background: #fff;
  width: ${screenWidth};
  height: ${screenHeight}
  padding: 50px;
  position: absolute;
  z-index: 100;
`;

export const ContentWrap = styled.View`
  padding-top: 20px;
`;
ContentWrap.displayName = 'ContentWrap';

const Content = styled.View`
  flex-direction: column;
`;

const TitleWrap = styled.Text`
  font-size: 20px;
`;

const TitleText = styled.Text`
  color: ${props => props.color};
`;

const DiaryText = styled.Text`
  margin: 10px 0 40px;
  font-size: 16px;
  color: #000;
`;

const Dot = styled.View`
  width: 20px;
  height: 20px;
  background: ${props => props.color};
`;

const CloseIconWrap = styled.View`
  color: #000;
  margin-left: auto;
`;

const AnimatedContainer = Animated.createAnimatedComponent(Container);
