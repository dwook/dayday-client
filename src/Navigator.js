import React from 'react';
import {Text, View, Button, SafeAreaView, TouchableOpacity} from 'react-native';
import HomeScreen from '../screens/HomeScreen';
import LoginScreen from '../screens/LoginScreen';
import UserScreen from '../screens/UserScreen';
import DiaryScreen from '../screens/DiaryScreen';
import TestScreen from '../screens/TestScreen';
import ModalScreen from '../screens/ModalScreen';
import AppLoadingScreen from '../screens/AppLoadingScreen';
import GoodScreen from '../screens/Diary/GoodScreen';
import BadScreen from '../screens/Diary/BadScreen';
import PlanScreen from '../screens/Diary/PlanScreen';
import {createAppContainer, createSwitchNavigator} from 'react-navigation';
import {
  createMaterialTopTabNavigator,
  MaterialTopTabBar,
} from 'react-navigation-tabs';
import {createStackNavigator} from 'react-navigation-stack';
import {HomeIcon, WriteIcon, MenuIcon, CloseIcon} from '../components/Icons';

import styled from 'styled-components';

const CloseIconWrap = styled.View`
  margin-left: auto;
  padding-right: 40px;
  font-size: 16px;
  color: #fff;
`;

function SafeAreaDiaryTabBar(props) {
  return (
    <SafeAreaView style={{backgroundColor: 'black'}}>
      <TouchableOpacity onPress={() => props.navigation.navigate('Home')}>
        <CloseIconWrap>
          <CloseIcon />
        </CloseIconWrap>
      </TouchableOpacity>
      <MaterialTopTabBar {...props} />
    </SafeAreaView>
  );
}

function SafeAreaBottomTabBar(props) {
  return (
    <SafeAreaView style={{backgroundColor: 'black'}}>
      <MaterialTopTabBar {...props} />
    </SafeAreaView>
  );
}

export const DiaryTabs = createMaterialTopTabNavigator(
  {
    Good: GoodScreen,
    Bad: BadScreen,
    Plan: PlanScreen,
  },
  {
    tabBarComponent: SafeAreaDiaryTabBar,
    tabBarOptions: {
      tabStyle: {
        height: 70,
        backgroundColor: 'black',
      },
      labelStyle: {
        fontSize: 20,
      },
    },
  },
);

export const DiaryStack = createStackNavigator(
  {
    DiaryTabs,
  },
  {
    headerMode: 'none',
  },
);

export const MainStack = createMaterialTopTabNavigator(
  {
    Home: {
      screen: HomeScreen,
      navigationOptions: {
        tabBarLabel: 'Home',
        tabBarIcon: ({focused}) => (
          <HomeIcon stroke={focused ? '#fff' : '#495057'} />
        ),
      },
    },
    Diary: {
      screen: View,
      navigationOptions: {
        tabBarOnPress: ({navigation}) => navigation.navigate('DiaryStack'),
        tabBarIcon: ({focused}) => (
          <WriteIcon stroke={focused ? '#fff' : '#495057'} />
        ),
      },
    },
    User: {
      screen: UserScreen,
      navigationOptions: {
        tabBarLabel: 'User',
        tabBarIcon: ({focused}) => (
          <MenuIcon stroke={focused ? '#fff' : '#495057'} />
        ),
      },
    },
  },
  {
    tabBarPosition: 'bottom',
    tabBarComponent: SafeAreaBottomTabBar,
    swipeEnabled: false,
    tabBarOptions: {
      showIcon: true,
      showLabel: false,
      tabStyle: {
        height: 70,
        backgroundColor: 'black',
      },
    },
  },
);

export const AppStack = createStackNavigator(
  {
    MainStack,
    DiaryStack,
  },
  {
    headerMode: 'none',
    mode: 'modal',
  },
);

export const AuthStack = createStackNavigator({Login: LoginScreen});

export default createAppContainer(
  createSwitchNavigator(
    {
      AppLoading: AppLoadingScreen,
      App: AppStack,
      Auth: AuthStack,
    },
    {
      initialRouteName: 'AppLoading',
    },
  ),
);
