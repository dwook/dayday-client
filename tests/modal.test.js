/**
 * @jest-environment jsdom
 */

import React from 'react';
import 'react-native';
import 'jest-enzyme';

import {configure, shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {AppContext} from '../src/Provider';
import 'jest-styled-components';
import 'styled-components-test-utils/lib/jest';

import Modal from '../components/Modal';

configure({adapter: new Adapter(), disableLifecycleMethods: true});

jest.mock('react-native-voice', () => {});

describe('Modal Component', () => {
  it('should match to snapshot', () => {
    const wrapper = shallow(<Modal />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should renders ContentWrap', () => {
    const wrapper = mount(
      <AppContext.Provider
        value={{modalTop: 0, selectedDiary: {good: '', bad: '', plan: ''}}}>
        <Modal />
      </AppContext.Provider>,
    );
    expect(wrapper.find('ContentWrap')).toHaveLength(1);
  });
});
