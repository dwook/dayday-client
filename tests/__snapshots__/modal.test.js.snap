// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Component should match to snapshot 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
